src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -29,11 +30,12 @@ internal static BoundBlock InjectInstrumentation(MethodSymbol method, BoundBlock
                     TypeSymbol payloadElementType = boolType;
                     ArrayTypeSymbol payloadType = ArrayTypeSymbol.CreateCSharpArray(compilation.Assembly, payloadElementType);
                     FieldSymbol payloadField = GetPayloadField(method, methodOrdinal, payloadType, factory);
+                    bool methodUsesArrowSyntax = methodBody.Syntax.Kind() == SyntaxKind.ArrowExpressionClause;
```

Comment:
The more general issue turned out to involve all methods without explicit blocks, so this particular naming issue evaporated.